server/src/test/java/org/elasticsearch/common/xcontent/builder/XContentBuilderTests.java

```diff
@@ -93,61 +95,61 @@ public void testRaw() throws IOException {
             XContentBuilder xContentBuilder = XContentFactory.contentBuilder(XContentType.JSON);
             xContentBuilder.startObject();
-            xContentBuilder.rawField("foo", new BytesArray("{\"test\":\"value\"}"));
+            xContentBuilder.rawField("foo", new BytesArray("{\"test\":\"value\"}").streamInput());
             xContentBuilder.endObject();
-            assertThat(xContentBuilder.bytes().utf8ToString(), equalTo("{\"foo\":{\"test\":\"value\"}}"));
+            assertThat(BytesReference.bytes(xContentBuilder).utf8ToString(), equalTo("{\"foo\":{\"test\":\"value\"}}"));
         }
         {
             XContentBuilder xContentBuilder = XContentFactory.contentBuilder(XContentType.JSON);
             xContentBuilder.startObject();
-            xContentBuilder.rawField("foo", new BytesArray("{\"test\":\"value\"}"));
-            xContentBuilder.rawField("foo1", new BytesArray("{\"test\":\"value\"}"));
+            xContentBuilder.rawField("foo", new BytesArray("{\"test\":\"value\"}").streamInput());
+            xContentBuilder.rawField("foo1", new BytesArray("{\"test\":\"value\"}").streamInput());
             xContentBuilder.endObject();
-            assertThat(xContentBuilder.bytes().utf8ToString(), equalTo("{\"foo\":{\"test\":\"value\"},\"foo1\":{\"test\":\"value\"}}"));
+            assertThat(BytesReference.bytes(xContentBuilder).utf8ToString(), equalTo("{\"foo\":{\"test\":\"value\"},\"foo1\":{\"test\":\"value\"}}"));
         }
         {
             XContentBuilder xContentBuilder = XContentFactory.contentBuilder(XContentType.JSON);
             xContentBuilder.startObject();
             xContentBuilder.field("test", "value");
-            xContentBuilder.rawField("foo", new BytesArray("{\"test\":\"value\"}"));
+            xContentBuilder.rawField("foo", new BytesArray("{\"test\":\"value\"}").streamInput());
             xContentBuilder.endObject();
-            assertThat(xContentBuilder.bytes().utf8ToString(), equalTo("{\"test\":\"value\",\"foo\":{\"test\":\"value\"}}"));
+            assertThat(BytesReference.bytes(xContentBuilder).utf8ToString(), equalTo("{\"test\":\"value\",\"foo\":{\"test\":\"value\"}}"));
         }
         {
             XContentBuilder xContentBuilder = XContentFactory.contentBuilder(XContentType.JSON);
             xContentBuilder.startObject();
             xContentBuilder.field("test", "value");
-            xContentBuilder.rawField("foo", new BytesArray("{\"test\":\"value\"}"));
+            xContentBuilder.rawField("foo", new BytesArray("{\"test\":\"value\"}").streamInput());
             xContentBuilder.field("test1", "value1");
             xContentBuilder.endObject();
-            assertThat(xContentBuilder.bytes().utf8ToString(), equalTo("{\"test\":\"value\",\"foo\":{\"test\":\"value\"},\"test1\":\"value1\"}"));
+            assertThat(BytesReference.bytes(xContentBuilder).utf8ToString(), equalTo("{\"test\":\"value\",\"foo\":{\"test\":\"value\"},\"test1\":\"value1\"}"));
```

Comment:
These should probably move into xcontent's tests rather than server's tests so it won't be able to use this.

But that can be another PR :).

It does make me wonder if `string` should be implemented in the xcontent artifact without Lucene.

Reply:
I think these should use `Strings.toString(xContentBuilder)` at least.